### ISSUE TITLE
How to embed Ammonite-REPL in Scala 3?

### ISSUE THREAD

**Issue by vaclavsvejcar** [com-lihaoyi/Ammonite#1250]
I used embedded _Ammonite-REPL_ with _Scala 2.x_ successfully, but after migrating to _Scala 3_ I'm having trouble to compile the code using the guide from [documentation](https://ammonite.io/#Embedding):

```scala
ThisBuild / scalaVersion := "3.1.1"
...
libraryDependencies += "com.lihaoyi" % "ammonite" % "2.5.2" cross CrossVersion.full
```

Then in code

```scala
ammonite.Main(predefCode = code).run("ast" -> ast, "defs" -> defs, "json" -> json)
```

and I'm getting compile error:

```
[error] 45 |      ammonite.Main(predefCode = code).run("ast" -> ast, "defs" -> defs, "json" -> json)
[error]    |                                           ^^^^^^^^^^^^
[error]    |                          Found:    (String, dev.svejcar.sjq.model.Node)
[error]    |                          Required: ammonite.util.Bind[?]
```

It looks that Scala 3 doesn't like either the implicit conversion in `ammonite.util.Bind#ammoniteReplArrowBinder` or how it uses the `TypeTag` stuff (which AFAIK is not in Scala 3).

My question is, is _Ammonite-REPL_ supposed to work embedded from Scala 3 code and if yes, what should I do to make it work?

**Comment by edwardcwang** [comment]
It's possible to work around the arrow binder issue but then we get stuck with "No TypeTag available for Int".

```scala
ammonite.Main().run(ammonite.util.Bind("myInt", 1234))
```

**Comment by lolgab** [comment]
This is fixed in `3.0.0-M0-59-cdeaa580`

**Comment by edwardcwang** [comment]
Confirmed working in `3.0.0-M0-59-cdeaa580` and Scala 3.3.1.
